spec/beaker/dsl/helpers/facter_helpers_spec.rb

```diff
@@ -0,0 +1,59 @@
+require 'spec_helper'
+
+class ClassMixedWithDSLHelpers
+  include Beaker::DSL::Helpers
+  include Beaker::DSL::Wrappers
+  include Beaker::DSL::Roles
+  include Beaker::DSL::Patterns
+
+  def logger
+    RSpec::Mocks::Double.new('logger').as_null_object
+  end
+
+end
+
+describe ClassMixedWithDSLHelpers do
+  let( :command ){ 'ls' }
+  let( :host )   { double.as_null_object }
+  let( :result ) { Beaker::Result.new( host, command ) }
+
+  let( :master ) { make_host( 'master',   :roles => %w( master agent default)    ) }
+  let( :agent )  { make_host( 'agent',    :roles => %w( agent )           ) }
+  let( :custom ) { make_host( 'custom',   :roles => %w( custom agent )    ) }
+  let( :dash )   { make_host( 'console',  :roles => %w( dashboard agent ) ) }
+  let( :db )     { make_host( 'db',       :roles => %w( database agent )  ) }
+  let( :hosts )  { [ master, agent, dash, db, custom ] }
+
+
+  describe '#fact_on' do
+    it 'retrieves a fact on a single host' do
+      result.stdout = "family\n"
```

Comment:
Not sure if these tests are new or just being moved around, so take with a grain of salt. These tests seem very stubby (i.e., they're maybe just white-box testing the fact that a thing is implemented by calling another thing).  I wonder if there's a way to make these tests less brittle and more valuable by integrating more with the real code-under-test?